### PR TITLE
Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by anderCM**
- Create class `Person` with the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`, `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `of_age?` that returns true if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Create class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(name, age = 'Unknown', permission: true)
```
Comment (prolajumokeoni) [inline person.rb:5]:
```suggestion
  def initialize(name, age = 'Unknown', permission: true)
  @id = Random.rand(1..1000)
```

- [x] Kindly add `id` in your initialized method it should be  a random integer between 1 and 1000 👍🏿

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+class Teacher < Person
+  def initialize(name, especialization, age = 'Unknown', permission: true)
+    super(name, age, permission: permission)
+    @especialization = especialization
```
Comment (prolajumokeoni) [inline teacher.rb]: - [x] It would be better to  delete the `e` before specialization, please fix it  👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @anderCM 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The classes to represent students, person and teachers have been implemented💪🏿
 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+class Teacspecializationher < Person
```
Comment (omar-muhamad) [inline teacher.rb]: - [x] great job making the required changes, but Kindly rename this class to `Teacher` again as per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=%22%C2%AF%5C(%E3%83%84)/%C2%AF%22.-,Create%20class%20Teacher%20with%20the%20following,-%3A).

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @anderCM,

#### Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: teacher.rb (hunk 3 above)
Comment (anderCM) [reply teacher.rb]: @Omar-Muhamad  It's weird because I didn't change it, but I'm going to rename it anyway 👍

**Review by victorokoroji — APPROVED** [review]
Hi @anderCM, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (anderCM) [reply person.rb:5]: thanks for the advice, I will ad the `id` for every object created

File: teacher.rb (hunk 2 above)
Comment (anderCM) [reply teacher.rb]: Thanks for the advice, I will remove the `e`

**Pull request merged**